File: class34/Anton_Georgiev_2.cc
#include <iostream>
using namespace std;

int power(long long int n, long long int p) {
	long long int nn=1;
	
	for(int i=1;i<=p;++i) {
		nn*=n;
	}
	return nn;
}

int main() {
	long long int n, d, e, m, c, dm;
	char check;
	
	cout << "Molq vavedete 'n': ";cin >> n;
	cout << "Molq vavedete 'e': ";cin >> e;
	cout << "Molq vavedete 'd': ";cin >> d;
	cout << "Molq vavedete suobshtenie: ";cin >> m;
	cout << "Molq izberete 'c' ili 'd': ";cin >> check;
	
	if(check=='c') {
		c=power(m, e)%n;
		cout << c << endl;
	} else if(check=='d') {
		dm=power(m, d)%n;
		cout << dm << endl;
	} else {
		cout << "Wrong choice!" << endl;
	}

	return 0;
}//34059

File: class35_homework/Anton_Georgiev_2.rb
require "digest/sha1"

print "Molq vuvedete suobshtenie: "
m=gets.chomp().to_i
print "Molq izberete RSA ili DSA: "
check=gets.chomp()

if check=="RSA"
	print "Molq vuvedete n: "
	n=gets.chomp().to_i
	print "Molq vuvedete e: "
	e=gets.chomp().to_i
	
	c=m**e%n
	puts c
elsif check=="DSA"
	print "Molq vuvedete p: "
	p=gets.chomp().to_i
	print "Molq vuvedete q: "
	q=gets.chomp().to_i
	print "Molq vuvedete g: "
	g=gets.chomp().to_i
	print "Molq vuvedete y: "
	y=gets.chomp().to_i
	print "Molq vuvedete x: "
	x=gets.chomp().to_i
	
	begin
		k=rand(q-1)+1
		r=(g**k%p)%q
		s=(((Digest::SHA1.digest(m.to_s)).to_i+x*r)/k)%q
	end until(s!=0 and r!=0)
	
	printf "%d, %d\n", r, s
end
